src/Workspaces/Core/Portable/SourceGeneration/IRemoteSourceGenerationService.cs

```diff
@@ -21,15 +21,15 @@ internal interface IRemoteSourceGenerationService
     /// compare that to the prior generated documents it has to see if it can reuse those directly, or if it needs to
     /// remove any documents no longer around, add any new documents, or change the contents of any existing documents.
     /// </summary>
-    ValueTask<ImmutableArray<(SourceGeneratedDocumentIdentity documentIdentity, SourceGeneratedDocumentContentIdentity contentIdentity)>> GetSourceGenerationInfoAsync(
+    ValueTask<ImmutableArray<(SourceGeneratedDocumentIdentity documentIdentity, SourceGeneratedDocumentContentIdentity contentIdentity, DateTime generationDateTime)>> GetSourceGenerationInfoAsync(
         Checksum solutionChecksum, ProjectId projectId, CancellationToken cancellationToken);
 
     /// <summary>
     /// Given a particular set of generated document ids, returns the fully generated content for those documents.
     /// Should only be called by the host for documents it does not know about, or documents whose checksum contents are
     /// different than the last time the document was queried.
     /// </summary>
-    ValueTask<ImmutableArray<(string contents, DateTime generationDateTime)>> GetContentsAsync(
+    ValueTask<ImmutableArray<string>> GetContentsAsync(
```

Comment:
this value moved from the content request (which is really just about getting the excat content for a checksum) to the above call to get info about a generated file. THat's not only better, it's more correct as we can def be in a situation where we say "we generated the same contents in OOP as last time, but the generation timestamps were incremented".